[quant] Enable QNNPACK tests for TestSerialization

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37351 [quant] Enable qnnpack tests for test_quantize and test_numeric_suite
* **#37263 [quant] Enable QNNPACK tests for TestSerialization**

Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21286402](https://our.internmc.facebook.com/intern/diff/D21286402)

supriyar committed:
[quant] Enable QNNPACK tests for TestSerialization
Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on aten/src/ATen/native/quantized/cpu/qnnpack/src/conv-run.cc:
```diff
@@ -281,6 +281,7 @@ struct QnnpackDeleter {
   }
 };
 
+__attribute__((no_sanitize("function")))
```
@kimishpatel seems like this is still causing the ASAN issue - https://circleci.com/gh/pytorch/pytorch/5266916?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

kimishpatel replied:
Sorry, little out of clue at this point. I think one thing can be tried is no_sanitize("undefined"). This should be a big hammer that bypasses UBSAN on this function. If that doesnt work either then probably we are not able to apply the attribute to the function. Best if you can repro and try these things locally.

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Enable QNNPACK tests for TestSerialization"


Summary:

Test Plan:
python test/test_quantization.py TestSerialization
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21286402](https://our.internmc.facebook.com/intern/diff/D21286402)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_backward_compatibility.py:
```diff
@@ -67,57 +71,47 @@ def remove_prefix(text, prefix):
-        self._test_op(module, input_size=[1, 3, 6, 6], generate=False)
-        # TODO: graph mode quantized conv2d module
+    @given(qengine=st.sampled_from(("qnnpack", "fbgemm")))
+    def test_linear(self, qengine):
```
when generate is True, we only want to do this once to generate the serialized data, could you refactor the code so that we only run it once when generate is True?

jerryzh168 commented on test/quantization/test_backward_compatibility.py:
```diff
@@ -67,57 +71,47 @@ def remove_prefix(text, prefix):
-                                 groups=1, bias=True, padding_mode="zeros")
-        self._test_op(module, input_size=[1, 3, 6, 6], generate=False)
-        # TODO: graph mode quantized conv2d module
+    @given(qengine=st.sampled_from(("qnnpack", "fbgemm")))
```
also, we don't need to use hypothesis here, looping our the engines is better
